### PR TITLE
Enhance ES index cleaner e2e test to verify indices have been removed

### PR DESCRIPTION
**Pull request by objectiser**
Existing test only verifies that the cronjob completes successfully.

This PR extends the test to verify that the job has correctly triggered the es-index-cleaner image and that the task has been performed correctly (i.e. the indices have been removed).

Signed-off-by: Gary Brown <gary@brownuk.com>

### CODE REVIEW

File: test/e2e/es_index_cleaner_test.go [hunk 1]
```diff
@@ -71,19 +77,47 @@ func esIndexCleanerTest(t *testing.T, f *framework.Framework, testCtx *framework
...
 	defer portForw.Close()
 	defer close(closeChan)
-	err = SmokeTest("http://localhost:16686/api/traces", "http://localhost:14268/api/traces", "foo-bar", retryInterval, timeout)
+
+	esPod, err := GetPod("default", "elasticsearch", "elasticsearch", f.KubeClient)
 	if err != nil {
 		return err
 	}
-
-	err = WaitForCronJob(t, f.KubeClient, namespace, fmt.Sprintf("%s-es-index-cleaner", name), retryInterval, timeout)
```
Comment (objectiser) [inline test/e2e/es_index_cleaner_test.go]: Currently removed the intermediate checks of whether the cronjob completed - as was focusing on the result. But wondering whether this should be added back in - just to check that the job also reports success.

File: test/e2e/es_index_cleaner_test.go [hunk 2]
```diff
@@ -71,19 +77,47 @@ func esIndexCleanerTest(t *testing.T, f *framework.Framework, testCtx *framework
...
+		return fmt.Errorf("jaeger-span index not found prior to es-index-cleaner: err = %v", err)
+	}
+
+	return wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		flag, err := hasIndex(t)
+		return !flag, err
+	})
+}
+
+func hasIndex(t *testing.T) (bool, error) {
```
Comment (pavolloffay) [inline test/e2e/es_index_cleaner_test.go]: why passing `t`? maybe it's  better to change the signature to `bool containsIndex(string) bool, error`

Comment (objectiser) [reply test/e2e/es_index_cleaner_test.go]: Yes, `t` no longer necessary so will remove. Will also update the signature and check for `jaeger-` for now.

File: test/e2e/es_index_cleaner_test.go [hunk 3]
```diff
@@ -85,5 +107,27 @@ func esIndexCleanerTest(t *testing.T, f *framework.Framework, testCtx *framework
 	if err != nil {
 		return err
 	}
-	return nil
+
+	return wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		flag, err := hasIndexWithPrefix("jaeger-")
```
Comment (pavolloffay) [inline test/e2e/es_index_cleaner_test.go]: Is this needed `hasIndexWithPrefix` is also called on line 97.

Comment (objectiser) [reply test/e2e/es_index_cleaner_test.go]: Line 97 is checking the indices exist, this part of checking they no longer exist.

**Comment by objectiser** [comment]
Still investigating the build failure in travis - the problem is not lint, but the `gosec` command performed in the lint target. I've not been able to reproduce the behaviour locally yet, but found odd behaviour with `gosec` - produces a different return status depending upon whether the `-quiet` option is provided (which should only affect what is output).

File: test/e2e/es_index_cleaner_test.go (hunk 3 above)
Comment (pavolloffay) [reply test/e2e/es_index_cleaner_test.go]: oh my bad, it looks good.

**Comment by objectiser** [comment]
I have created a separate issue #379 for the `gosec` problem and currently disabled it, as it was not a direct result of this PR (out of memory problem). So will merge this PR.

**Pull request merged**